src/server/controllers/users.js

```diff
@@ -128,5 +134,82 @@ module.exports = {
         errorLogs.catchErrors(error);
         res.status(400).send(error);
       });
+  },
+  inviteUser(req, res) {
+    const name = getUsernameFromEmail(req.body.email);
+    req.body.username = name.first + ' ' + name.last;
+    const userObject = {
+      email: req.body.email,
+      username: req.body.username,
+      roleId: req.body.roleId,
+      locationId: req.body.locationId
+    };
+    return User.findOrCreate({where: { email: userObject.email},
+      defaults: userObject})
+      .spread( async (createdUser, created) => {
+        if(!created) {
+          return res.status(409).send({ message: 'User already exists' });
```

Comment:
If a user already exists, update their role to march the role in the request to ensure those users with the role 1 e.g reporter, witness and are PnC members who  are being made assignees or admins have their roleId updated appropriately.